src/atomate2/common/schemas/magnetism.py

```diff
@@ -0,0 +1,207 @@
+    in the MagneticOrderingOutput and MagneticOrderingRelaxation documents.
+    """
+
+    structure: Structure = Field(None, description="Input structure")
+    ordering: Ordering = Field(
+        None,
+        description=(
+            "The magnetic ordering of the input structure, "
+            "as defined in pymatgen.analysis.magnetism.analyzer."
+        ),
+    )
+    symmetry: str = Field(None, description="Detected space group symbol.")
+
+
+class MagneticOrderingRelaxation(BaseModel):
+    """
+    Defines the relaxation information for a magnetic ordering calculation. This is
+    embedded within the MagneticOrderingOutput.
+    """
+
+    uuid: str = Field(None, description="Unique ID of the calculation.")
+    dir_name: str = Field(None, description="Directory of the calculation.")
+    input: MagneticOrderingInput = Field(
+        None, description="Input ordering information."
+    )
+    structure: Structure = Field(
+        None, description="Final structure from the calculation."
+    )
+    symmetry_changed: bool = Field(None, description="Whether or not symmetry changed.")
+    ordering_changed: bool = Field(
+        None,
+        description=(
+            "Specifies whether or not the magnetic ordering changed during the"
+            " calculation."
+        ),
+    )
+    ordering: Ordering = Field(None, description="Final ordering from the calculation.")
+    symmetry: str = Field(None, description="Detected space group symbol.")
+    energy: float = Field(None, description="Final energy result from the calculation.")
+    energy_per_atom: float = Field(None, description="Final energy per atom.")
+
+    @classmethod
+    def from_task_document(
+        cls, task_document, uuid: str | None = None
+    ) -> MagneticOrderingOutput:
+        """
+        Construct a MagneticOrderingRelaxation output doc from a task document. This is
+        to be implemented for the DFT code of choice.
+        """
+        raise NotImplementedError
+
+
+class MagneticOrderingOutput(BaseModel):
+    """
+    Defines the output for a *static* magnetic ordering calculation. This is used
+    within the construction of the MagneticOrderingDocument. If a relaxation was
+    performed, this information will be stored within the relax_output field.
+    """
+
+    uuid: str = Field(None, description="Unique ID of the calculation.")
+    dir_name: str = Field(None, description="Directory of the calculation.")
+    input: MagneticOrderingInput = Field(
+        None, description="Input ordering information."
+    )
+    structure: Structure = Field(
+        None, description="Final structure from the calculation."
+    )
+    ordering: Ordering = Field(
+        None,
+        description=(
+            "The magnetic ordering of the output structure, "
+            "as defined in pymatgen.analysis.magnetism.analyzer."
+        ),
+    )
+    magmoms: list[float] = Field(None, description="Magnetic moments of the structure.")
```

Comment:
Not a request to modify, but adding a note for reference: it would be useful to round magnetic moments somewhat, since getting a primitive structure (with respect to the magnetic moments, i.e. without losing magnetic order) will not work as intended unless equivalent moments are exactly equal. Typically, the noise in the moment is sufficient such that this is not the case. I did not get around to adding this however.

Reply:
Should we enable `round_magmoms=True` in the `CollinearMagneticStructureAnalyzer`?

Reply:
Yes, I'd do that. Eventually the underlying algorithm in `round_magmoms` should be updated however. Simple rounding (e.g. `np.round`) will sometimes round two similar magmoms in opposite directions. I tried instead doing a KDE of the magmoms with a suitable Gaussian width and taking the peak positions, but this felt like it was over-complicating things!